File: ex9.go
import "math"

func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    palindrome, n, i := 0, 0, 1
    for cpy := x; cpy != 0; cpy /= 10 {
        n += 1
    }
    i = n - 1
    for cpy := x; cpy != 0; cpy /= 10 {
        digit := cpy % 10
        palindrome += digit * int(math.Pow10(i))
        i -= 1
    }
    return palindrome == x
}

File: leetcode/ex9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0))
            return false;
        int r = 0;
        while (x > r) {
            r = 10*r + x % 10;
            x /= 10;
        }
        return x == r || x == r/10;
    }
};
